flink-optimizer/src/main/java/org/apache/flink/optimizer/plan/Channel.java

```diff
@@ -134,6 +134,10 @@ public Channel(PlanNode sourceNode, TempMode tempMode) {
 	public PlanNode getSource() {
 		return this.source;
 	}
+
+	public void setSource(PlanNode source) {
```

Comment:
I think we do not need this method, if we create completely new channels in the `RangePartitionRewriter`.